android/src/main/java/com/mattermost/networkclient/NetworkClient.kt

```diff
@@ -75,6 +78,18 @@ internal class NetworkClient(private val context: ReactApplicationContext, priva
             applyClientBuilderConfiguration(options, cookieJar)
         }
 
+        val fingerprintsMap = getCertificatesFingerPrints()
+        if (fingerprintsMap.isNotEmpty()) {
+            val pinner = CertificatePinner.Builder()
+            for ((domain, fingerprints) in fingerprintsMap) {
+                for (fingerprint in fingerprints) {
+                    pinner.add(domain, "sha256/$fingerprint")
+                }
+            }
```

Comment:
I was having issues doing it the same way as in Android creating in this case a serverTrustManager for each client.. but this will change hopefully with the new changes I'm introducing. If not there is no real harm in doing it this way, just that it may have more certs loaded in memory than needed for a given client

Reply:
no.. check getCertificatesFingerPrints function, this does check the domain using the class baseUrl property